awsx/role.ts

```diff
@@ -69,7 +69,7 @@ export function defaultRoleWithPolicies(
 ): {
   roleArn?: pulumi.Output<string>;
   role?: aws.iam.Role;
-  policies?: aws.iam.RolePolicyAttachment[];
+  policies?: pulumi.Output<aws.iam.RolePolicyAttachment[]>;
```

Comment:
I thought there was some magic way to do this change for the SDKs while still receiving un-wrapped values in the provider but I may be mistaken. If this works it's good. 

I slightly worry about provisioning resources under apply inside the component resource, but I might need some platform team help to find references to how this works. 

https://www.pulumi.com/docs/concepts/resources/components/#registering-component-outputs

Normally there is a call registerOutputs at the end of the constructor.

We don't seem to be calling it for ec2TaskDefinition:

https://github.com/pulumi/pulumi-awsx/blob/master/awsx/ecs/ec2TaskDefinition.ts#L32

If we're doing apply then children may register out-of-order that is after the component resource is done constructing and I vaguely recall this may cause bugs.